include/frameAndLink.h

```diff
@@ -0,0 +1,92 @@
+#ifndef FRAMEANDLINK_H
+#define FRAMEANDLINK_H
+
+#include "typedefs.h"
+#include "transformationBuffer.h"
+#include "stampedTransformation.h"
+
+#include <string>
+#include <vector>
+#include <QtGui/QQuaternion>
+
+class Frame;
+
+/***************************
+ * LINK                    *
+ ***************************/
+
+class Link
+{
+
+friend class GMLWriter;
+
+public:
+
+    Link(Frame *const _parent, Frame *const _child, const DurationSec &_storageTime)
+    : parent{_parent}
+    , child(_child)
+    , buf(_storageTime)
+    {}
+
+    bool oldestTransformation(const FrameID &srcFrame, StampedTransformation &stampedTransformation);
+
+    bool newestTransformation(const FrameID &srcFrame, StampedTransformation &stampedTransformation);
+
+    bool addTransformation(const FrameID &srcFrame, StampedTransformation stampedTransformation);
+
+    bool transformationAtTimeT(const FrameID &srcFrame, StampedTransformation &stampedTransformation);
+
+    bool distanceToNextClosestEntry(const Timestamp &tStamp, std::chrono::milliseconds &distanceToCloserEntry);
+
+    Frame *const parent;
+
+    Frame *const child;
+
+    double weight = 1.;
+
+    void writeJSON(QJsonObject &json) const;
+
+protected:
+
+    enum AccessType { TIME, OLDEST, NEWEST };
+
+    bool getTransformation(const FrameID &srcFrame, StampedTransformation &stampedTransformation, AccessType accessType);
+
+    void invertTransformation(StampedTransformation &stampedTransformation);
+
+     TransformationBuffer buf;
+};
+
+
+/***************************
+ * FRAME                   *
+ ***************************/
+
+class Frame
+{
+public:
+
+    Frame(const FrameID &_frameID)
+    : frameID(_frameID)
+    {}
+
+    void addLink(Link* const newLink);
+
+    void connectionTo(const FrameID &destination, Link* &linkToDest);
```

Comment:
Simply returning the pointer would make more sense. I decided to do it that way to be consistent, since I return everything through the argument.

Done.